astropy/modeling/core.py

```diff
@@ -801,6 +802,76 @@ def inverse(self):
         raise NotImplementedError("An analytical inverse transform has not "
                                   "been implemented for this model.")
 
+    @property
+    def bounding_box(cls):
+        """
+        Returns a tuple defining the bounding box limits in each input 
+        dimension, symmetric about the position, 
+        e.g. ``((y_low, y_high), (x_low, x_high))``
+
+        Input must be tuple or array-like with shape (model.n_inputs, 2) 
+        or (2 * model.n_inputs,)
+
+        ``model.bounding_box = (-5, 5, -5, 5)``
+
+        is equivalent to
+
+        ``model.bounding_box = ((-5, 5), (-5, 5))``
+
+        ``model.bbox = 'auto'`` will set the default value defined in the
+        model class.
+
+        ``model.bbox = None`` will turn off the bounding box altogether
+        """
+
+        if cls._bounding_box == 'auto':
+            return cls._bounding_box_default()
+        else:
+            return cls._bounding_box
+
+    @bounding_box.setter
+    def bounding_box(cls, limits):
+        """
+        Assigns the bounding box limits  e.g. ``((y_low, y_high), (x_low, x_high))``
+        Input must be tuple or array-like with shape (model.n_inputs, 2) 
+        or (2 * model.n_inputs,)
+
+        ``model.bounding_box = (-5, 5, -5, 5)``
+
+        is equivalent to
+
+        ``model.bounding_box = ((-5, 5), (-5, 5))``
+
+        ``model.bbox = 'auto'`` will set the default value defined in the
+        model class.
+
+        ``model.bbox = None`` will turn off the bounding box altogether
+        """
+
+        nd = cls.n_inputs
+
+        try:
+            if limits in ('auto', None):
+                pass
+
+            elif nd == 1:
+                assert np.shape(limits) == (2,)
+                limits = tuple(limits)
+
+            elif np.shape(limits) == (2 * nd,):
+                limits = np.array(limits).reshape(nd, 2)
+                limits = tuple([tuple(lim) for lim in limits])
+
+            else:
+                assert np.shape(limits) == (nd, 2)
+                limits = tuple([tuple(lim) for lim in limits])
+
+        except AssertionError:
+            "Bounding box must be tuple or array-like of shape \
```

Comment:
I guess you wanted this string to be printed?
Currently as far as I can see the string is just an expression that's not assigned to anything, i.e. has no effect.

@patti – Please try triggering this in interactive use, and eventually add a test for this.

I think the correct pattern is to raise another error (not sure of which error class here, like `AssertionError` or `ValueError`), like this:
```python
try:
   ...
except AssertionError:
   raise ValueError('message')
```